Checks/Compose/PatternLengthCheck.cs

```diff
@@ -0,0 +1,219 @@
+using MapsetVerifierFramework.objects.metadata;
+
+using MVTaikoChecks.Utils;
+
+using static MVTaikoChecks.Global;
+using static MVTaikoChecks.Aliases.Difficulty;
+using static MVTaikoChecks.Aliases.Mode;
+using static MVTaikoChecks.Aliases.Level;
+
+namespace MVTaikoChecks.Checks.Compose
+{
+    [Check]
+    public class PatternLengthCheck : BeatmapSetCheck
+    {
+        private const string _WARNING = nameof(_WARNING);
+        private const bool _DEBUG_SEE_ALL_PATTERN_LENGTHS = false;
+
+        private readonly Beatmap.Difficulty[] _DIFFICULTIES = new Beatmap.Difficulty[]
+        {
+            DIFF_KANTAN,
+            DIFF_FUTSUU,
+            DIFF_MUZU,
+            DIFF_ONI
+        };
+
+        public override CheckMetadata GetMetadata() =>
+            new BeatmapCheckMetadata()
+            {
+                Author = "SN707",
+                Category = "Compose",
+                Message = "Pattern Lengths",
+                Difficulties = _DIFFICULTIES,
+                Modes = new Beatmap.Mode[] { MODE_TAIKO },
+                Documentation = new Dictionary<string, string>()
+                {
+                    {
+                        "Purpose",
+                        @"
+                    Preventing small snaps from being too long based on each difficulty's Ranking Criteria."
+                    },
+                    {
+                        "Reasoning",
+                        @"
+                    Certain snaps if going on for too long are too straining for certain difficulties."
+                    }
+                }
+            };
+
+        public override Dictionary<string, IssueTemplate> GetTemplates() => new Dictionary<string, IssueTemplate>()
+        {
+
+            {
+                _WARNING,
+                new IssueTemplate(LEVEL_WARNING,
+                    "{0} {1} {2} pattern is {3} notes long, ensure this makes sense",
+                    "start", "end", "snap", "number")
+                .WithCause("Chain length is surpassing the RC guideline")
+            }
+        };
+
+        public override IEnumerable<Issue> GetIssues(BeatmapSet beatmapSet)
+        {
+
+            // Difficulty, <snap size, snap count>
+            var shortSnapParams = new Dictionary<Beatmap.Difficulty, Dictionary<double, int>>()
+            {
+                { DIFF_KANTAN, new Dictionary<double, int>() {
+                    { 1.0 / 1, 7 },
+                    { 1.0 / 2, 2 }
+                }},
+                { DIFF_FUTSUU, new Dictionary< double, int>() {
+                    { 1.0 / 2, isBottomDiffKantan(beatmapSet) ? 7 : 5 }, // If no kantan, then recommended maximum 1/2 length for futsuu is 5 instead of 7
+                    { 1.0 / 3, 2 }
+                }},
+                { DIFF_MUZU, new Dictionary<double, int>() {
+                    { 1.0 / 4, 5 }, 
+                    { 1.0 / 6, 4 }, 
+                }},
+                { DIFF_ONI, new Dictionary<double, int>() {
```

Comment:
Oni is missing a 1/6 pattern length check which seems unintentional. best to limit the length to 4 just like muzu, because anything longer will 99.9% get modded out lol

Reply:
Seems reasonable, added